drivers: i3c: Fix NXP driver consuming a lot of CPU time

mmahadevan108 opened the pull request:
The I3C and I2C transactions are consuming a lot of CPU time. The stated polling check was is not completed in sub-microseconds as is stated in the comment.
Added a 1usec sleep between probes of status register.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/91134

PetervdPerk-NXP commented on drivers/i3c/i3c_mcux.c:
```diff
@@ -140,10 +139,9 @@ static int reg32_poll_timeout(volatile uint32_t *reg,
+	 * quickly. Add a small delay between checks
 	 */
-	if (!WAIT_FOR((sys_read32((mm_reg_t)reg) & mask) == match, timeout_us, /*nop*/)) {
+	if (!WAIT_FOR((sys_read32((mm_reg_t)reg) & mask) == match, timeout_us, k_usleep(1))) {
```
We're trading CPU load for latency, and technically it could happen with k_usleep() that FIFO buffer underflows.
I would recommend implementing this using interrupts instead removing the need for active polling.

mmahadevan108 committed:
drivers: i3c: Update NXP drivers transfer function

Change the transfer implementation to use interrupts
instead of a busy wait loop.
Added a Kconfig to specify a timeout period to wait
for a transfer to complete.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

mmahadevan108 committed:
drivers: i3c_mcux: Update error handling

Enable the Error interrupt to capture errors and process
them accordingly.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

dleach02 commented:
@mmahadevan108 what do you want to do with this PR?
